common/src/main/java/earth/terrarium/heracles/api/quests/defaults/ItemQuestIcon.java

```diff
@@ -4,27 +4,38 @@
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.teamresourceful.resourcefullib.client.scissor.ScissorBoxStack;
 import earth.terrarium.heracles.Heracles;
+import earth.terrarium.heracles.api.client.WidgetUtils;
 import earth.terrarium.heracles.api.quests.QuestIcon;
 import earth.terrarium.heracles.api.quests.QuestIconType;
 import net.minecraft.client.gui.GuiGraphics;
 import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.item.Item;
+import net.minecraft.world.item.Items;
 
 public record ItemQuestIcon(Item item) implements QuestIcon<ItemQuestIcon> {
 
     public static final QuestIconType<ItemQuestIcon> TYPE = new Type();
 
     @Override
     public void render(GuiGraphics graphics, ScissorBoxStack scissor, int x, int y, int width, int height) {
-        graphics.renderFakeItem(item.getDefaultInstance(), x + (width - 16) / 2, y + (height - 16) / 2);
+        if (height == 32 && width == 32) { // Blatant ugly workaround. Centering should be handled better in both cases.
+            WidgetUtils.drawItemIcon(graphics, item.getDefaultInstance(), x, y, width);
+        } else {
+            graphics.renderFakeItem(item.getDefaultInstance(), x + (width - 16) / 2, y + (height - 16) / 2);
+        }
     }
 
     @Override
     public QuestIconType<ItemQuestIcon> type() {
         return TYPE;
     }
 
+    @Override
+    public boolean isVisible() {
+        return item != null && item != Items.AIR;
```

Comment:
I would perfer instead of `!=` for the item comparison that `item.is` is used as to future proof.

Reply:
I'm not familiar with any method called `is()` on item - which one of us is confused?

Reply:
Sorry I meant `equals` `is` is only on the `ItemStack`

Reply:
there is not an itemstack in this expression at all, there are two items, one is constant - am I missing something?

Reply:
What I meant is my bad about the `is` method as I realized that it was only on the `ItemStack` here you should be using `!Items.AIR.equals(item)` instead of `!=`

```suggestion
        return item != null && !Items.AIR.equals(item);
```

Reply:
This doesn't make a difference as it's a reference check regardless, however you can also do `Object.equals(item, AIR)` and remove the null check entirely.

Reply:
uh, can't we not do that considering it's an inequality check?

Reply:
Oh true, my bad.

Reply:
We've factored out `isVisible` and made `render` just always return boolean, which simplifies things a bit - this null / non-air check exists at the top of ItemQuestIcon.render now.